subsys/bluetooth/mesh/dfd_srv.c

```diff
@@ -1049,3 +1144,87 @@ enum bt_mesh_dfd_status bt_mesh_dfd_srv_fw_delete_all(struct bt_mesh_dfd_srv *sr
 
 	return BT_MESH_DFD_SUCCESS;
 }
+
+int bt_mesh_dfd_srv_oob_check_complete(struct bt_mesh_dfd_srv *srv,
+				       const struct bt_mesh_dfu_slot *slot, int status,
+				       uint8_t *fwid, size_t fwid_len)
+{
+	struct bt_mesh_dfu_slot *existing_slot;
+	int idx;
+
+	if (slot != srv->upload.slot || !srv->upload.is_oob) {
```

Comment:
I don't see the need to do that. This check is to stop the case of Upload Start(slot1) and then Check Complete(slot2), for instance. To abort if the application somehow completes the check for a slot that's not the current upload slot (which as mentioned should not happen with a well-behaved application implementation).

If I understand you correctly, you are worried about what would happen in this case:

  * Upload Start OOB (slot1)
  * App calls check complete (slot1)
  * Upload finishes or fails
  * Malformed app calls check complete (slot1) again for some reason, while `srv->upload.slot` still points to slot1

?

In that case, it will simply return `-EALDREADY` on line 1167. It will keep returning this until a new `srv->upload.slot` is set by one of the other functions in this module.

Reply:
The `is_oob` check is to just return early here if an application for some weird reason tries to call check complete on a slot which is currently participating in an _in-band_ transfer